packages/html/ds/src/card/card.schema.ts

```diff
@@ -1,16 +1,17 @@
 import * as zod from 'zod';
+import { buttonSchema } from '../button/button-schema';
 import { iconSchema } from '../icon/icon.schema';
+import { linkSchema } from '../link/link.schema';
+import { tagSchema } from '../tag/tag.schema';
 
-const actionSchema = zod.object({
-  href: zod.string({
-    description: 'URL for the action link',
-    required_error: 'href is required for actions',
+const actionSchema = zod.union([
+  buttonSchema.extend({
+    type: zod.literal('button').describe('Type of action is a button'),
   }),
-  text: zod.string({
-    description: 'Text for the action link',
-    required_error: 'text is required for actions',
+  linkSchema.extend({
+    type: zod.literal('link').describe('Type of action is a link'),
   }),
-});
+]);
 
 export const cardSchema = zod.object({
   type: zod.enum(['vertical', 'horizontal'], {
```

Comment:
I believe you should replace `zod.enum` to `zod.nativeEnum`, otherwise it will raise an error while building `html/ds`

Type 'string[]' is not assignable to type 'Record<string, string>'.
  Index signature for type 'string' is missing in type 'string[]'